src/stress_analyzer.py

```diff
@@ -0,0 +1,157 @@
+import argparse
+from remote_execution import *
+
+### Functions below are for getting initial container dictionaries
+
+# Function that calls on the correct get_container function based on TYPE
+def get_container_ids(vm_ips, services, stress_type):
+    if stress_type == 'ALL':
+        return get_container_ids_all(vm_ips, services)
+    elif stress_type == 'BINARY':
+        return get_container_ids_binary(vm_ips, services)
+
+
+# Returns a dictionary of the services and their respective container_ids
+# VM_IPS is a list of ip addresses, and SERVICES is a list of services
+def get_container_ids_all(vm_ips, services):
+    container_id_dict = {}
+    for vm_ip in vm_ips:
+        ssh_client = quilt_ssh(vm_ip)
+        docker_container_id = 'docker ps | tr -s \' \' | cut -d \' \' -f1 | tail -n +2'
+        docker_container_cmd = 'docker ps --no-trunc | grep -oP \'\"\K[^\"]+(?=[\"])\''
+        _, stdout1, _ = ssh_client.exec_command(docker_container_id)
+        container_ids = stdout1.read().splitlines()
+        _, stdout2, _ = ssh_client.exec_command(docker_container_cmd)
+        container_commands = stdout2.read().splitlines()
+        for i in range(len(container_ids)):
+            if services == '*' or (container_commands[i] in services):
+                container_id_dict[container_commands[i]] = (vm_ip, container_ids[i])
```

Comment:
Won't this just remember that last machine that is iterated through?
Say I have service A running on Vm0 and Vm1. 

Container_id_dict would only save one of the tuples, no?